modules/core/src/main/java/org/apache/ignite/internal/processors/security/IgniteSecurityProcessor.java

```diff
@@ -100,7 +105,14 @@ public IgniteSecurityProcessor(GridKernalContext ctx, GridSecurityProcessor secP
     @Override public OperationSecurityContext withContext(SecurityContext secCtx) {
         assert secCtx != null;
 
-        secPrc.touch(secCtx);
+        try {
+            secPrc.touch(secCtx);
+        }
+        catch (SecurityException e) {
+            // Security context is not propagated yet.
+            // Log this warning and proceed an operation with already given security context.
+            log.warning("Failed to check security context [subj=" + secCtx.subject().id() + ", err=" + e + ']');
```

Comment:
Let's change to

```suggestion
            log.warning("Failed to wait for security context propagation [subj=" + secCtx.subject().id() + ", err=" + e + ']');
```

because we don't really "check" the context (we already have it), we just wait for it to be written.

Reply:
Actually, it would also be great to add the propagation timeout here. Perhaps we need to do this in the private part?

Reply:
> because we don't really "check" the context (we already have it), we just wait for it to be written.
Security context propagation is a detail of internal implementation. AI security knows nothing about that.
Javadoc states the following:
```
    /**
     * This method is called before an attempt to do any secure operation using given context.
     *
     * Implementation can check if a context still valid, refresh internal state or do some other
     * implementation specific things.
     *
     * @param secCtx Security context.
     * @throws SecurityException If a context is no longer valid.
     */
    public default void touch(SecurityContext secCtx) throws SecurityException {
        // No-op.
    }
```

Perhaps, you're right and this change should be moved to the private part.